Update VOISS-Net procedure

darren-tpk opened the pull request:
Growing interest in VOISS-Net has prompted requests to run the model on locally stored data rather than pulling from Earthscope. Previously, functions such as `check_timeline` coupled data acquisition (via `waveform_collection`'s `gather_waveforms`) with spectrogram classification. This coupling made it difficult to use VOISS-Net on waveforms not accessible through an FDSN `Client`.

This update refactors several functions to decouple data retrieval from classification. Key functions like `check_timeline` now operate directly on an `ObsPy` `Stream` object. Users can construct this Stream by various means, but we primarily provide tutorial scripts that cover both FDSN `Client` sources (`timeline_checker.py`) and local MiniSEED directories (`timeline_checker_local.py`).

We also performed some minor debugging, and tidied `toolbox.py` for additional clarity.

davidfee5 committed:
Revise citations and publication details in README

Updated references and citations in the README file to reflect the latest publication details.

darren-tpk committed:
Update toolbox.py

- Remove load_data (we will rely on waveform_collection's read_local moving forward)
- Add download_data function (to save FDSN client waveforms into IRIS-compliant miniseeds)
- Fixed waveform scaling factor for plot_spectrogram
- Deprecate use of "denominator" to format x-ticks. Plotting codes now allow matplotlib to decide on where x-ticks should be.

darren-tpk committed:
Remove old pavlof rsam code

darren-tpk committed:
Remove rotate_NE_to_RT

Delete obsolete function, use obspy's rotate_ne_rt

darren-tpk committed:
Update taper_length argument in process_waveform

follows obspy tapering convention

darren-tpk committed:
Fix download_data to add .mseed file format and change coord to coords

darren-tpk committed:
Fix download_data's output coords.json to match read_local's requirements

darren-tpk committed:
Update check_timeline function

decoupling stream pulling utility from function

darren-tpk committed:
Add warnings import and update download_data to use filepaths

darren-tpk committed:
Update timeline_checker tutorial script

Decouples waveform pull from check_timeline

darren-tpk committed:
Add timeline_checker_local tutorial script

darren-tpk committed:
Better code comments

darren-tpk committed:
Better code comments

darren-tpk committed:
Tidy tutorial scripts

darren-tpk committed:
Remove sort_sta_distance import

We now use tr.stats coordinates information to sort

darren-tpk committed:
check_timeline tidy up

darren-tpk committed:
Update check_timeline_binned to use input stream.

darren-tpk committed:
Rename variables in plot_timeline_binned

darren-tpk committed:
Clean up

darren-tpk committed:
Use read_local_update branch in waveform_collection

darren-tpk committed:
Update README.md

darren-tpk committed:
Update quotation marks

darren-tpk committed:
Update generate_timeline_indicators to use data_keys

darren-tpk committed:
minor fix in plot_timeline

darren-tpk committed:
Tidy up plot_timeline and plot_timeline_binned, make sure voted result allows pnorm thresholding

darren-tpk committed:
Add some print statements to plot_timeline and plot_timeline_binned

darren-tpk committed:
fix trace RR on plot_spectrogram, minor figure and typo fixes

davidfee5 commented:
This is looking good to me and I've tested most of it out. What do you think about including the station distance in `check_timeline` if available? I think it is more useful than the channel name.

darren-tpk committed:
Tidy up white space and add fi_kwargs into example script

darren-tpk committed:
Label source-station distance if volc_lat and volc_lon provided to check_timeline

darren-tpk committed:
add fi_kwargs option to timeline_checker_local script

darren-tpk committed:
small quotation mark fix for consistency

darren-tpk commented:
Thanks for the reminder @davidfee5 ! `check_timeline` now checks if `volc_lat` and `volc_lon` are provided in `dr_kwargs` or `fi_kwargs`. If so, it will add the source-station distance to the spectrogram y-label, and if not, it will fall back to labeling channel name.

I also decided to not sort the station order internally within `check_timeline`, just in case the user wants a certain station with high SNR to appear first, for instance. Note that distance-sorting is currently done just before `check_timeline` is called:
```
# Sort stream by distance to volcano (order of the stream input determines order of subplots)
stream.traces.sort(key=lambda tr: gps2dist_azimuth(VOLC_COORDS[0], VOLC_COORDS[1], tr.stats.latitude, tr.stats.longitude)[0])
```

Lastly, I added `FI_KWARGS` as an option to both `timeline_checker` and `timeline_checker_local` scripts for awareness. I think it has been underutilized since we added it to the code.

davidfee5 committed:
change to wavelength

davidfee5 committed:
remove unused functions

davidfee5 reviewed (approved):
Looks good to me! I didn't test our the `read_local` much but everything I saw looks improved.

darren-tpk commented:
Awesome! Thanks for reviewing quickly.
